File: src/domain/Product_Detail.ts
export default class Product_Detail {
  constructor(
    public weight: number,
    public dimensions: object,
    public materials: string,
    public country_of_origin: string,
    public warranty_info: string,
    public care_instructions: string,
    public additional_info: object,
    public id?: number,
  ) {}
}

File: preonboarding/src/main/java/com/example/preonboarding/domain/ProductDetail.java
package com.example.preonboarding.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "product_detail")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class ProductDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    private Products products;

    private double weight;

    @Column(columnDefinition = "json")
    private String dimensions;

    private String materials;
    private String countryOfOrigin;
    private String warrantyInfo;
    private String careInstructions;

    @Column(columnDefinition = "jsonb")
    private String additionalInfo;
}
